src/test/java/com/google/sps/ServletUtilsTest.java

```diff
@@ -0,0 +1,131 @@
+package com.google.sps;
+
+import static com.google.sps.utils.StringConstants.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.TestUtils;
+import com.google.sps.utils.ServletUtils;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public final class ServletUtilsTest extends Mockito {
+  private final String TESTING_PARAMETER = "language";
+  private final String TESTING_PARAMETER_VALUE = "english";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  
+  private DatastoreService datastore;
+  private Entity groupEntityExpected;
+  private PrintWriter printWriter;
+  private StringWriter stringWriter;
+
+  @Mock
+  HttpServletRequest request;
+
+  @Mock
+  HttpServletResponse response;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+
+    groupEntityExpected = TestUtils.createGroupEntity(this.datastore);
```

Comment:
Does this need to happen in the setUp()? or can you put it in the variable declaration above?

Reply:
stringWriter does not, I can just flush it every time before a test. printWriter does, since it depends on stringWriter. I have to declare datastore every time (because of the way that testing datastore works), so I have to create the groupEntityExpected in the setUp() every time.